Add codecov token to AZP

uppittu11 opened the pull request:
### PR Summary:
I am attempting to make `codecov` upload coverage report to codecov.io from AZP yml.
### PR Checklist
------------
 - [ ] Includes appropriate unit test(s)
 - [ ] Appropriate docstring(s) are added/updated
 - [ ] Code is (approximately) PEP8 compliant
 - [ ] Issue(s) raised/addressed?

uppittu11 commented:
It looks like the Linux+Python3.7 build uploaded a coverage report to codecov and updated the codecov comment as intended. However, the title of the report [here](https://codecov.io/gh/mosdef-hub/mbuild/commit/c6e45819423ba6c393ad43d522faa86aa1d69317) doesn't seem right.

Let's hold off on merging this for the moment

mattwthompson commented:
What are the chances it is tracked if this is merged into master and it takes effect on the next PR? Is there any way to test this short of doing that? I'm fine with doing this if we think it's the best option. I forget if @umesh-timalsina weight on this specifically earlier in Slack?

uppittu11 commented:
I think everything is working correctly now. The latest commits removed the codecov upload step from travis and the most recent edits to the codecov comment above were generated via azp.

uppittu11 commented:
Looks like we can also upload the codecov results to azure pipelines, so it will display the coverage there ([see here](https://blog.codecov.io/2019/01/24/new-in-codecov-4-4-3/)). I think it will look something like this:
![image](https://user-images.githubusercontent.com/33552857/66931969-7da0d480-effc-11e9-9004-bab83580efdc.png)

Not sure if this is something we care to add.

mattwthompson commented:
Looks nice!

uppittu11 commented:
Ok. Here is what the `Tests` and `Coverage` tabs look like in Azure: [CI Results](https://dev.azure.com/mosdef/mosdef/_build/results?buildId=45&view=ms.vss-test-web.build-test-results-tab)

The tests detects that we have 7 runs, but it lumps them all together so we have 3129 tests in total, which is kind of annoying, but not really a problem. 

The coverage tab doesn't follow our codecov yaml, so it includes the `tests/` and `examples/` directories in the report. That's probably the reason it says we have `51.7%` coverage. However, codecov's website, status checks, and comments have the correct numbers so we still get the green check mark on GitHub. 

I'll look into setting ignored directories in the coverage report.

umesh-timalsina commented:
@uppittu11 this [page](https://dev.azure.com/mosdef/mosdef/_build/results?buildId=45&view=ms.vss-test-web.build-test-results-tab) is pretty darn cool. LGTM.

uppittu11 commented:
One more thing: I added lines to omit the `mbuild/examples` and `mbuild/tests` directories in the coverage report so now the codecov.io results should agree with the report on azure

uppittu11 commented:
One more small change haha. 

When you install `pytest-azurepipelines` (which I had added to `requirements-dev.txt`) on your local machine it clutters the stdout with some extraneous `#vso` logging messages. So I removed that package from `requirements-dev.txt` and added a couple of lines to `azure-pipelines.yml` such that the VMs install the package before they run `pytest`.

mattwthompson commented:
![image](https://user-images.githubusercontent.com/7935382/67093445-502c6600-f177-11e9-8358-e77eb4011855.png)

Why is there this ghost AZP build? The other 7 pass and the most recent build on AZP's site is 7/7 passing: https://dev.azure.com/mosdef/mosdef/_build/results?buildId=58

uppittu11 commented:
https://github.com/mosdef-hub/foyer/pull/273#issuecomment-543459579

uppittu11 commented:
@mattwthompson can you restart the Travis job? Looks like one of the builds stalled out. 

Otherwise, is there a reason we are waiting to merge this PR?

mattwthompson commented on azure-pipelines.yml:
```diff
@@ -1,8 +1,11 @@
 trigger:
-  - "*"
+  - master
```
Isn't this the opposite of what we just did in foyer?

uppittu11 replied:
I don't think we changed this in foyer. 

I reverted back to using `master` as the trigger based on our chat with @umesh-timalsina on slack, since whatever issues we had seemed to be resolved on his branch (https://github.com/umesh-timalsina/mbuild/pull/2)

umesh-timalsina replied:
[This File](https://github.com/umesh-timalsina/mbuild/blob/631-azp-fix/azure-pipelines.yml) adds PR trigger against master (Opposed to *). Updates `conda` to the latest version available (With a CI Fix from pandas https://github.com/pandas-dev/pandas/commit/74bbbb084dc530080da93b96a41114d0598f928b). On my fork, its working for PR, Scheduled builds are not failing.
